pkg/operator/controller/machineconfig/nodes.go

```diff
@@ -90,7 +90,7 @@ func (r *MachineConfigReconciler) ensureReqNodeLabelExists(ctx context.Context)
 
 	updNodeCount := 0
 	nodeList := &corev1.NodeList{}
-	if err := r.listWorkerNodes(ctx, nodeList); err != nil {
+	if err := r.listLabeledWorkerNodes(ctx, nodeList); err != nil {
```

Comment:
```suggestion
	if err := r.listNodes(ctx, nodeList, r.CtrlConfig.Spec.NodeSelector); err != nil {
```
This would avoid the need of additional method.